### PR TITLE
replace singular statement ++/-- with += 1/-= 1 for integer variables

### PR DESCRIPTION
**Pull request by dduan**
I busted out sed and replaced all the one-line ++ and -- statements in Swift files, then filtered out the results gradually until all tests passed in "check-swift-all".

Please let me know if breaking this up to multiple commit/PR is more desirable.

### CODE REVIEW

File: stdlib/private/StdlibUnittest/StdlibCoreExtras.swift [hunk 1]
```diff
@@ -57,8 +57,8 @@ func findSubstring(string: String, _ substring: String) -> String.Index? {
       }
       if needle[needleIndex] == haystack[matchIndex] {
         // keep advancing through both the string and search string on match
-        ++matchIndex
-        ++needleIndex
+        matchIndex += 1
+        needleIndex += 1
```
Comment (gribozavr) [inline stdlib/private/StdlibUnittest/StdlibCoreExtras.swift]: This part does not compile on Linux:

```
swift/stdlib/private/StdlibUnittest/StdlibCoreExtras.swift:60:20: error: binary operator '+=' cannot be applied to operands of type 'String.UnicodeScalarView.Index' and 'Int'
```

**Comment by gribozavr** [comment]
@dduan Could you fix the compilation issue on Linux? Just omitting that part of the patch should be fine. Please squash patches after that fix.

@nadavrot I have reviewed the patch, and it only affects variables typed as integers.  Do you have performance concerns with this kind of change?

**Comment by dabrahams** [comment]
When advancing something that is primarily an index, `x = x.successor()` may be more appropriate than `x += 1`  even when you happen to know it's an integer. Please consider each case carefully; thanks.

**Comment by dduan** [comment]
@gribozavr done.

**Comment by dduan** [comment]
@dabrahams Ahah! It just happens so that after sed, I had to leave a lot of ++/-- statement unchanged because they are for Index types, which can't be operand of +=/-=. You just give me an idea for the next patch :P

**Comment by nadavrot** [comment]
@gribozavr Thanks for checking with me.  We'll track the performance of this change after it lands. I don't have any objections for committing this patch.

**Comment by falcon03** [comment]
Is there any reason for the removal of these (extremely handy) operators, other than the (extremely debatable IMHO) ones exposed at

https://github.com/apple/swift-evolution/blob/master/proposals/0004-remove-pre-post-inc-decrement.md

Not to mention the fact that we’re not sure wether
let x = 1
x += 1

can have the same performance as
let x = 1
x++

(and the same applies for the — operator)

If discussion can still be reopened I’d like to share my opinions on the “disadvantages” exposed in the previously linked change proposal.



> Il giorno 16 dic 2015, alle ore 02:40, Dmitri Gribenko <notifications@github.com> ha scritto:
> 
> Merged #556 <https://github.com/apple/swift/pull/556>.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/apple/swift/pull/556#event-492965231>.
> 


 
 
 --
 ZE-Light e ZE-Pro: servizi zimbra per caselle con dominio email.it, per tutti i dettagli 
Clicca qui http://posta.email.it/caselle-di-posta-z-email-it/?utm_campaign=email_Zimbra_102014=main_footer/f
 
 Sponsor:
 Soluzioni di email hosting per tutte le esigenze: dalle caselle gratuite a quelle professionali su piattaforma Zimbra, da quelle su proprio dominio a quelle certificate PEC. Confronta le soluzioni
 Clicca qui: http://adv.email.it/cgi-bin/foclick.cgi?mid=13325&d=16-12

**Comment by gribozavr** [comment]
@falcon03 You are welcome to send your arguments to swift-evolution@swift.org.

**Pull request merged**